app/src/main/java/org/oppia/android/app/administratorcontrols/AdministratorControlsActivity.kt

```diff
@@ -86,6 +88,46 @@ class AdministratorControlsActivity :
       isProfileDeletionDialogVisible
     )
     title = resourceHandler.getStringInLocale(R.string.administrator_controls)
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+      onBackInvokedDispatcher.registerOnBackInvokedCallback(1000) {
+        val fragment =
+          supportFragmentManager.findFragmentById(
+            R.id.administrator_controls_fragment_multipane_placeholder
+          )
+        /*
+         * If the current fragment is ProfileListFragment then the activity should end on back press.
+         * If it's instead ProfileEditFragment then profileListFragment should be inflated via
+         * handleOnBackPressed.
+         */
+        if (fragment is ProfileEditFragment) {
+          administratorControlsActivityPresenter.handleOnBackPressed()
+        } else {
+          onBackPressedDispatcher.onBackPressed()
+        }
+      }
+    } else {
```

Comment:
I propose removing this bit of the onBackInvokedDispatcher.registerOnBackInvokedCallback() and remaining with the part that is under "else".

According to the documentation, https://developer.android.com/reference/android/app/Activity#onBackPressed(), `onBackPressedDispatcher.addCallback()` is backward compatible, eliminating the need to gate the logic with the sdk version.